src/main/java/webserver/http/request/HttpRequestUrl.java

```diff
@@ -0,0 +1,71 @@
+package webserver.http.request;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import webserver.http.exception.InvalidHttpRequestException;
+
+public class HttpRequestUrl {
+    private static final String PATH_AND_PARAMETER_DIVIDER = "?";
+    private static final String PARAMETERS_DIVIDER = "&";
+    private static final String PARAMETER_DIVIDER = "=";
+    private static final String EMPTY_CONTENT = "";
+
+    private final String requestUrl;
+    private final Map<String, String> parameters = new LinkedHashMap<>();
+
+    public HttpRequestUrl(String requestUrl) {
+        this.requestUrl = requestUrl;
+        String parameterLine = requestUrl.substring(requestUrl.indexOf(PATH_AND_PARAMETER_DIVIDER) + 1);
+        parseParameter(requestUrl, parameterLine);
+    }
+
+    private void parseParameter(String requestUrl, String parameterLine) {
+        if (requestUrl.contains(PATH_AND_PARAMETER_DIVIDER)) {
+            String[] splitedParameters = parameterLine.split(PARAMETERS_DIVIDER);
+
+            for (String splitedParameter : splitedParameters) {
+                String[] keyAndValue = splitedParameter.split(PARAMETER_DIVIDER);
+                parameters.put(keyAndValue[0], keyAndValue[1]);
+            }
+            validate(parameters, splitedParameters.length);
+        }
+    }
+
+    private void validate(Map<String, String> parameters, int originalLength) {
+        if (parameters.size() != originalLength) {
+            throw new InvalidHttpRequestException("겹치는 key가 있습니다.");
+        }
+
+        if (parameters.containsKey(EMPTY_CONTENT)) {
+            throw new InvalidHttpRequestException("정보가 없는 key가 있습니다.");
+        }
+    }
+
+    public String getPath() {
+        if (hasParameters()) {
+            return requestUrl.substring(0, requestUrl.indexOf(PATH_AND_PARAMETER_DIVIDER));
+        }
+        return requestUrl;
+    }
+
+    public boolean isSamePath(String path) {
+        return getPath().equals(path);
+    }
+
+    private boolean hasParameters() {
+        return requestUrl.contains(PATH_AND_PARAMETER_DIVIDER);
+    }
```

Comment:
parameters map 내의 데이터 존재 유무가 더 정확하지 않을까요. :)

Reply:
```suggestion
    private boolean hasParameters() {
        return !parameters.isEmpty();
    }
```

허허.. 맞네요 화투!

생각이 짧았습니다.

수정하였습니다🙂